tests/taggers/test_classification_inference.py

```diff
@@ -0,0 +1,463 @@
+# Copyright 2025 The Marin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import tempfile
+
+import fsspec
+import pytest
+import ray
+
+from marin.core.runtime import TaskConfig
+from marin.processing.classification.config.inference_config import DatasetSchemaConfig
+from marin.processing.classification.config.inference_config import InferenceConfig, RuntimeConfig
+from marin.processing.classification.inference import (
+    process_file_with_quality_classifier_streaming,
+    read_dataset_streaming,
+    run_inference,
+)
+from marin.utils import fsspec_exists, fsspec_mkdirs
+
+DEFAULT_DATASET_SCHEMA = DatasetSchemaConfig(input_columns=["id", "text"], output_columns=["id", "attributes"])
+
+
+@pytest.fixture
+def sample_data():
+    """Create sample data for testing"""
+    return [
+        {"id": "doc1", "text": "This is a high quality document with excellent content."},
+        {"id": "doc2", "text": "This is an average document with some useful information."},
+        {"id": "doc3", "text": "This is a poor quality document with minimal content."},
+        {"id": "doc4", "text": "Another excellent document with comprehensive details."},
+        {"id": "doc5", "text": "A mediocre document that could be improved."},
+    ]
+
+
+@pytest.fixture
+def temp_dir():
```

Comment:
nit, pytest has this builtin: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture